### PR TITLE
refactor(text-to-ast): convert to typescript 

### PR DESCRIPTION
**Pull request by anoriqq**
- Convert `@textlint/text-to-ast` to TypeScript code

Ref. #669

### CODE REVIEW

File: packages/@textlint/text-to-ast/src/plaintext-parser.ts [hunk 1]
```diff
@@ -72,7 +73,7 @@ function createBRNode(lineNumber, startIndex) {
  * @param {[TxtNode]} nodes
  * @returns {TxtNode} Paragraph node
  */
-function createParagraph(nodes) {
+function createParagraph(nodes: [TxtNode]): TxtNode {
```
Comment (azu) [inline packages/@textlint/text-to-ast/src/plaintext-parser.ts:76]:
```suggestion
function createParagraph(nodes: TxtNode[]): TxtNode {
```

`TxtNode[]` is Array of TxtNode.
`[TxtNode]` is an item in a array.

File: packages/@textlint/text-to-ast/src/plaintext-syntax.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
 // LICENSE : MIT
 "use strict";
-const { ASTNodeTypes } = require("@textlint/ast-node-types");
-const exports = {
+import { ASTNodeTypes } from "@textlint/ast-node-types";
+
+export default {
```
Comment (azu) [inline packages/@textlint/text-to-ast/src/plaintext-syntax.ts:5]: Can you use named exports as possible?

```suggestion
export const Syntax = {
```

Also need to change to `import { Syntax }`

Comment (azu) [reply packages/@textlint/text-to-ast/src/plaintext-syntax.ts:5]: https://basarat.gitbook.io/typescript/main-1/defaultisbad

File: packages/@textlint/text-to-ast/src/plaintext-parser.ts (hunk 1 above)
Comment (anoriqq) [reply packages/@textlint/text-to-ast/src/plaintext-parser.ts]: I also fixed jsdoc.

**Comment by azu** [comment]
Thanks!

**Pull request merged**